modules/navier_stokes/src/postprocessors/PressureDrop.C

```diff
@@ -0,0 +1,339 @@
+#ifdef NDEBUG
+      if (status_known)
+        break;
+#endif
+    }
+    // we ll assume all face infos for a given element and side have the same upstream status
+    // it seems very unlikely that upstream and downstream boundaries would be touching
+    // and the delimitation would be within a refined element's face
+    if (status_known)
+      break;
+  }
+
+  if (upstream)
+  {
+    // Integration loops are different for FE and FV at this point
+    if (_qp_integration)
+    {
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        _weighted_pressure_upstream +=
+            _JxW[_qp] * _coord[_qp] * computeQpWeightedPressureIntegral();
+        _weight_upstream += _JxW[_qp] * _coord[_qp] * computeQpWeightIntegral();
+      }
+    }
+    else
+    {
+      for (auto & fi : _face_infos)
+      {
+        _weighted_pressure_upstream +=
+            fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightedPressureIntegral(fi);
+        _weight_upstream += fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightIntegral(fi);
+      }
+    }
+  }
+  // Downstream contributions
+  else
+  {
+    if (_qp_integration)
+    {
+      for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+      {
+        _weighted_pressure_downstream +=
+            _JxW[_qp] * _coord[_qp] * computeQpWeightedPressureIntegral();
+        _weight_downstream += _JxW[_qp] * _coord[_qp] * computeQpWeightIntegral();
+      }
+    }
+    else
+    {
+      for (auto & fi : _face_infos)
+      {
+        _weighted_pressure_downstream +=
+            fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightedPressureIntegral(fi);
+        _weight_downstream += fi->faceArea() * fi->faceCoord() * computeFaceInfoWeightIntegral(fi);
+      }
+    }
+  }
+}
+
+Real
+PressureDrop::computeFaceInfoWeightedPressureIntegral(const FaceInfo * const fi) const
+{
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  mooseAssert(fi, "We should have a face info in " + name());
+
+  const bool correct_skewness =
+      _weight_interp_method == Moose::FV::InterpMethod::SkewCorrectedAverage;
+  auto elem_arg = std::make_tuple(_current_elem, _qp, _qrule);
+  mooseAssert(_qp == 0, "Only one quadrature point");
+
+  if (_weighting_functor)
+  {
+    const auto ssf =
+        Moose::FaceArg({fi,
+                        Moose::FV::limiterType(_weight_interp_method),
+                        MetaPhysicL::raw_value((*_weighting_functor)(elem_arg)) * fi->normal() > 0,
+                        correct_skewness,
+                        nullptr});
+    const auto face_weighting = MetaPhysicL::raw_value((*_weighting_functor)(ssf));
+    return fi->normal() * face_weighting * _pressure(ssf);
+  }
+  else
+  {
+    // External faces for the pressure
+    if (!fi->neighborPtr() || !_pressure.hasBlocks(fi->neighborPtr()->subdomain_id()))
+    {
+      const auto ssf = Moose::FaceArg(
+          {fi, limiterType(_weight_interp_method), true, correct_skewness, _current_elem});
```

Comment:
Are you indicating sidedness here on purpose, e.g. if we are on an internal face, then you want to ensure you don't use data from the other side? That would be the appropriate use of the sided-element argument. But then if you are going to do that ... are there ever going to be times where `!_pressure.hasBlocks(fi->elemPtr()->subdomain_id())` ?

Reply:
I'm mostly wondering why you've kept the `!fi->neighborPtr() || !_pressure.hasBlocks(fi->neighborPtr()->subdomain_id())` condition around here when you've made the `if (_weighting_functor)` branch nice and "clean" (by just always passing `nullptr` to the sided element argument)

Reply:
I think I misunderstood something critical about the FaceArg until now. I m going to adapt the docstring for it

Reply:
and you can tell me if the misunderstanding persists

Reply:
the addition to the doc-string looks good